akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -193,7 +193,7 @@ object Framing {
         } else if (buffer.slice(possibleMatchPos, possibleMatchPos + separatorBytes.size) == separatorBytes) {
           // Found a match
           val parsedFrame = buffer.slice(0, possibleMatchPos).compact
-          buffer = buffer.drop(possibleMatchPos + separatorBytes.size)
+          buffer = buffer.drop(possibleMatchPos + separatorBytes.size).compact
```

Comment:
Good idea, though let's keep it simple for now. Unless someone finds this dominating their usecase let's keep it as is. Properly doing that would require a FramingAttributes etc, a bit much work for here/right now